clang/test/SemaCXX/warn-self-move.cpp

```diff
@@ -35,11 +41,14 @@ class field_test {
   int x;
   field_test(field_test&& other) {
     x = std::move(x);  // expected-warning{{explicitly moving}}
+    x = static_cast<int&&>(x);  // expected-warning{{explicitly moving}}
```

Comment:
Can you add some tests where we should not get diagnostics for example like the one right below but w/ `static_cast`